Marlin/src/pins/ramps/pins_MKS_GEN_L.h

```diff
@@ -37,10 +39,17 @@
 // Heaters / Fans
 //
 
-#define MOSFET_A_PIN                          10  // HE0
-#define MOSFET_B_PIN                           7  // HE1 or FAN Hotend Cooling
-#define MOSFET_C_PIN                           8  // HBED
-#define FAN0_PIN                               9  // FAN Part Cooling
```

Comment:
Either the MOSFET pin scheme needs to be retained, or this file needs to do the handling of the flags `*_EFB`, `*_EEB`, etc., for cases where the Extruder, Fan, Bed arrangement may change. These flags may already be set for the `HAS_MULTI_HOTEND` and/or `HEATERS_PARALLEL` and should be preferred over referring to these settings flags directly.

Reply:
Reminder that the rule for Extruder, Fan, Bed pins is not to always define them according to their silkscreen label, but according to the configuration requirements. The board may say something like "H1, Fan, Bed" which would correspond to the common default `FET_ORDER_EFB`. But when two extruders are defined, the `FET_ORDER_EEB` flag gets set, so the pins should then be defined as "H1, H2, Bed" with no fan.

If the `FET_ORDER_*` flags aren't coming out right for this board according to those flags, either these FETs are just not defined in the correct sequence, or the general FET assignment in `pins_RAMPS.h` is wonky.

Reply:
Hmm, I wonder if it work just using the order 10, 9, 8, 7… But we have to look at the machines these ship in also and see if they are using non-standard ordering on shipping units.

Reply:
> Either the MOSFET pin scheme needs to be retained, or this file needs to do the handling of the flags *_EFB, *_EEB, etc.,

I really don't follow all that `FET_ORDER` stuff since we don't do that on non-AVR boards, but I have physically traced the pins on the MCU and verified that they now match real hardware.

Reply:
> Hmm, I wonder if it work just using the order 10, 9, 8, 7… But we have to look at the machines these ship in also and see if they are using non-standard ordering on shipping units.

We host several Artillery configs and the older/non-32-bit printers use a standard Gen-L: https://github.com/MarlinFirmware/Configurations/tree/import-2.1.x/config/examples/Artillery